sympy/integrals/tests/test_integrals.py

```diff
@@ -1170,3 +1170,11 @@ def test_issue_4950():
 
 def test_issue_4968():
     assert integrate(sin(log(x**2))) == x*sin(2*log(x))/5 - 2*x*cos(2*log(x))/5
+
+def test_singularities():
+    assert integrate(1/x**2,(x,-oo,oo)) == oo
```

Comment:
I timed the following functions : `1/x**2`  ,  `1/(x - 1)**2` and `1/(x**2 - 3*x + 2)` for both `singularities()` and `solveset()`
The `solveset()` call is much faster with more complex functions.

Code : 
```
from sympy import *
from sympy.calculus.singularities import singularities
import timeit

def func_singularities():
    x = Symbol('x')
    list(singularities(1/(x**2 - 3*x + 2),x))

def func_solveset():
    x = Symbol('x')
    list(solveset((1/(x**2 - 3*x + 2)).cancel().as_numer_denom()[1], x))

print(timeit.timeit(func_singularities,number=100))
print(timeit.timeit(func_solveset,number=100))
```
These were the times : 
```
1/(x**2)
('Singularities : ', 2.698566198348999)
('Solveset : ', 2.153296947479248)

1/((x-1)**2)
('Singularities : ', 3.8249380588531494)
('Solveset : ', 3.1464860439300537)

1/(x**2 - 3*x + 2)
('Singularities : ', 5.078230857849121)
('Solveset : ', 3.4630041122436523)
```